### PR TITLE
File Upload OnChangeRegister Events

### PR DESCRIPTION
**Pull request by RupalKotakDxc**
RMA-109598

### CODE REVIEW

File: projects/dxc-ngx-cdk/src/lib/dxc-file-input/dxc-file-input.component.ts [hunk 1]
```diff
@@ -262,8 +262,8 @@ export class DxcFileInputComponent
   onTouched: () => void = () => { };
   onChangeRegister = (val) => { };
 
-  writeValue(fileNames: any): void {
-    this.renderedValue = fileNames || "";
+  writeValue(fileData: any): void {
+    this.renderedValue = fileData || "";
```
Comment (assureclaims) [inline projects/dxc-ngx-cdk/src/lib/dxc-file-input/dxc-file-input.component.ts:266]: use fileData || null; instead fileData || "";

File: projects/dxc-ngx-cdk/src/lib/dxc-file-input/dxc-file-input.component.ts [hunk 2]
```diff
@@ -337,7 +337,7 @@ export class DxcFileInputComponent
   checkFileSize(file: File) {
     if(file.name!=null){
       let fileExtension = file.name.split('.').pop();
-      if(!this.accept.includes(fileExtension)){
+      if (!this.accept.toLocaleLowerCase().includes(fileExtension.toLocaleLowerCase())){
```
Comment (assureclaims) [inline projects/dxc-ngx-cdk/src/lib/dxc-file-input/dxc-file-input.component.ts:340]: toLowerCase instead of localelowercase

**Review by assureclaims — CHANGES_REQUESTED** [review]

File: projects/dxc-ngx-cdk/src/lib/dxc-file-input/dxc-file-input.component.ts (hunk 1 above)
Comment (RupalKotakDxc) [reply projects/dxc-ngx-cdk/src/lib/dxc-file-input/dxc-file-input.component.ts]: done

File: projects/dxc-ngx-cdk/src/lib/dxc-file-input/dxc-file-input.component.ts [hunk 3]
```diff
@@ -871,6 +871,7 @@ export class DxcFileInputComponent
     }
     this.fileService.delete(this.requests.removeRequest.url, fileData).pipe(take(1)).subscribe((response) => {
       const arr: FileData[] = [];
+      this.onChangeRegister({ files: arr, eventType: EventType.REMOVE });
```
Comment (assureclaims) [inline projects/dxc-ngx-cdk/src/lib/dxc-file-input/dxc-file-input.component.ts:874]: eventType: EventType.REMOVE  this is not required as we are able to check the previous value of control and new value and compare it. hence remove this and change it accordingly.

**Review by assureclaims — CHANGES_REQUESTED** [review]

File: projects/dxc-ngx-cdk/src/lib/dxc-file-input/dxc-file-input.component.ts (hunk 3 above)
Comment (RupalKotakDxc) [reply projects/dxc-ngx-cdk/src/lib/dxc-file-input/dxc-file-input.component.ts:874]: value change subscription at consuming service is getting called again and again until complete remove event is not succss so we need some identifier to understand that the remove event is emitted

**Review by assureclaims — APPROVED** [review]

File: projects/dxc-ngx-cdk/src/lib/dxc-file-input/dxc-file-input.component.ts (hunk 3 above)
Comment (assureclaims) [reply projects/dxc-ngx-cdk/src/lib/dxc-file-input/dxc-file-input.component.ts:874]: Please check with whom changes its calling again and again. for right now i am merging the code. Please check and fixed it

**Pull request merged**